### ISSUE TITLE
mapping table on snmp plugin use only first number to match

### ISSUE THREAD

**Issue by Truji** [influxdata/telegraf#961]
I'm using the mapping_table but looks like that is matching only with the first number of ifIndex value.
All instances are the same if the ifIndex value starts with the same number.

Example: 
> ifIndex,host=testing.lan,instance=dsc ifIndex=542i 1459782103301846509
> ifIndex,host=testing.lan,instance=dsc ifIndex=5i 1459782103304193032
> ifIndex,host=testing.lan,instance=dsc ifIndex=521i 1459782103313242647

Only ifIndex=5 is true: IF-MIB::ifDescr.5 = STRING: dsc

> ifIndex,host=testing.lan,instance=lo0 ifIndex=607i 1459782408362813308
> ifIndex,host=testing.lan,instance=lo0 ifIndex=6i 1459782408380025023
> ifIndex,host=testing.lan,instance=lo0 ifIndex=604i 1459782408381047861

Only ifIndex=6 is true: IF-MIB::ifDescr.6 = STRING: lo0

**Comment by titilambert** [comment]
Hello !
Could you give me your configuration ?
Thanks !

**Comment by Truji** [comment]
Yes, sorry.

```
[[inputs.snmp]] 
  snmptranslate_file = "/etc/telegraf/oids.txt"
  [[inputs.snmp.host]]
    address = "testing.lan:161"
    community = "public" # default public
    version = 2 # default 2
    timeout = 2.0 # default 2.0
    retries = 2 # default 2
    [[inputs.snmp.host.table]]
      name = "iftable1"
      include_instances = []
  [[inputs.snmp.table]]
    name = "iftable1"
    mapping_table =".1.3.6.1.2.1.2.2.1.2"
    sub_tables=[".1.3.6.1.2.1.2.2.1.1"]
```

**Comment by titilambert** [comment]
Thanks !
Could you run a snmpwalk on .1.3.6.1.2.1.2.2.1.1 and .1.3.6.1.2.1.2.2.1.2 ?

**Comment by Truji** [comment]
Attached, we have a few interfaces.

[.1.3.6.1.2.1.2.2.1.1.txt](https://github.com/influxdata/telegraf/files/202918/default.1.3.6.1.2.1.2.2.1.1.txt)
[.1.3.6.1.2.1.2.2.1.2.txt](https://github.com/influxdata/telegraf/files/202919/default.1.3.6.1.2.1.2.2.1.2.txt)

**Comment by Truji** [comment]
Hi, I looked again this and I can't see a pattern about. The problem is tagging the output, because if I use the include_instance the plugin filter correctly.

And is not happening every time, as you can see here:

> root@evee:~# telegraf -config /etc/telegraf/telegraf.conf -test | grep 563i
> ifIndex,host=testing.lan,instance=dsc ifIndex=563i 1459802458842428140
> root@grafana:~# telegraf -config /etc/telegraf/telegraf.conf -test | grep 563i
> ifIndex,host=testing.lan,instance=pe-0/1/10 ifIndex=563i 1459802460241818426

**Comment by titilambert** [comment]
This seems related to #1043

**Comment by sparrc** [comment]
this is now available in 1.0 RC 1 with the new SNMP plugin, please see here to download the latest version: https://github.com/influxdata/telegraf#linux-deb-and-rpm-packages, and [the changelog](https://github.com/influxdata/telegraf/blob/master/CHANGELOG.md) for details on the new SNMP plugin.
